README.md

```diff
@@ -109,6 +109,56 @@ PS C:\> $agentVersion = "v1.20.4"
 PS C:\> Initialize-ECSAgent -Cluster 'windows' -EnableTaskIAMRole -Version $agentVersion
 ```
 
+## Build ECS Agent from source
+
+### Build ECS Agent Image (Linux)
+
+ECS Agent can also be built locally from source on a linux machine. Use the following steps to build ECS Agent
+* Get ECS Agent source
+```
+git clone https://github.com/aws/amazon-ecs-agent.git
+```
+* Build Agent image using ```release-agent``` make target
+```
+make release-agent
+```
+This installs the required build dependencies, builds ECS Agent image and saves it at a path ```ecs-agent-v${AGENT_VERSION}.tar```. Load this using
+```
+docker load < ecs-agent-v${AGENT_VERSION}.tar
+```
+Follow the instructions [above](https://github.com/aws/amazon-ecs-agent#on-other-linux-amis) to continue with the installation
+
+### Build and run standalone (Linux)
+
+The Amazon ECS Container Agent may also be run outside of a Docker container as a Go binary. At this time, this is not recommended
+for production on Linux, but it can be useful for development or easier integration with your local Go tools.
+
+The following commands run the agent outside of Docker:
+
+```
+make gobuild
+./out/amazon-ecs-agent
+```
+
+### Standalone (Windows)
+
+The Amazon ECS Container Agent may be built by invoking `scripts\build_agent.ps1`
+
+### Scripts (Windows)
+
+The following scripts are available to help develop the Amazon ECS Container Agent on Windows:
+
+* `scripts\run-integ-tests.ps1` - Runs all integration tests in the `engine` and `stats` packages
+* `misc\windows-deploy\Install-ECSAgent.ps1` - Install the ECS agent as a Windows service
+* `misc\windows-deploy\amazon-ecs-agent.ps1` - Helper script to set up the host and run the agent as a process
+* `misc\windows-deploy\user-data.ps1` - Sample user-data that can be used with the Windows Server 2016 with Containers
+  AMI to run the agent as a process
+
+
+### Build ECS-Init Package (Linux)
+
+You can also build the ECS-Init packaged as a deb or rpm depending on the linux system you running. Follow instructions at [generic-deb-integrated](https://github.com/aws/amazon-ecs-agent/tree/master/packaging/generic-deb-integrated/debian) and [generic-rpm-integrated](https://github.com/aws/amazon-ecs-agent/tree/master/packaging/generic-rpm-integrated) to build and install ECS Agent packaged with Init deb or rpm packages
```

Comment:
nit: ECS-Init package?

Reply:
I think it can also mean as - ECS-Init can be packaged as a deb or rpm. But I can reword it is unclear.

Reply:
I get it now, thank you:)

Reply:
Updating in #3555